packages/core/upload/server/services/folder.js

```diff
@@ -1,32 +1,33 @@
 'use strict';
 
-const uuid = require('uuid').v4;
 const { keys, sortBy, omit, map, isUndefined } = require('lodash/fp');
 const { joinBy, setCreatorFields } = require('@strapi/utils');
 const { FOLDER_MODEL_UID, FILE_MODEL_UID } = require('../constants');
 const { getService } = require('../utils');
 
-const generateUID = () => uuid();
+const setPathIdAndPath = async folder => {
+  const { pathId: maxPathId } = await strapi.db
+    .queryBuilder(FOLDER_MODEL_UID)
+    .max('pathId')
+    .execute();
 
-const setPathAndUID = async folder => {
-  const uid = generateUID();
+  const pathId = maxPathId + 1;
```

Comment:
I'm wondering if this could lead to concurrency problems, if two users create a folder at the same time? Before the uid was always unique, where as now, it is dependent on the previous db state.

Maybe that's just the tradeoff of the approach ...

Reply:
I added a unique index in the database for this column, so it won't be possible to have duplicates. If 2 requests are done at the same time, one will fail then. Regarding how we create folders we should keep database consistency.